### PR TITLE
added highlighting `Jira` links

### PR DESCRIPTION
**Pull request by Hidanio**

### CODE REVIEW

File: src/main/kotlin/com/vk/admstorm/AdmStartupService.kt [hunk 1]
```diff
@@ -122,6 +125,23 @@ class AdmStartupService(private var project: Project) {
         })
     }
 
+    private fun setupIssueTrackers(project: Project) {
+        fun List<com.vk.admstorm.env.Service>.getByKey(key: String) = firstOrNull { it.key == key }
+
+        val prj = IssueNavigationConfiguration.getInstance(project)
```
Comment (Danil42Russia) [inline src/main/kotlin/com/vk/admstorm/AdmStartupService.kt:131]: replace `prj` with `navigationConfig`

File: src/main/kotlin/com/vk/admstorm/AdmStartupService.kt [hunk 2]
```diff
@@ -122,6 +125,23 @@ class AdmStartupService(private var project: Project) {
...
+        val service = Env.data.services.getByKey("jira") ?: return
+        val url = service.url + "/browse/\$0"
+
+        val isExist = prj.links.find { it ->
```
[the remark this answers is not in the document]
Comment (Hidanio) [reply src/main/kotlin/com/vk/admstorm/AdmStartupService.kt]: Done

File: src/main/kotlin/com/vk/admstorm/AdmStartupService.kt [hunk 3]
```diff
@@ -122,6 +126,21 @@ class AdmStartupService(private var project: Project) {
         })
     }
 
+    private fun setupIssueTrackers(project: Project) {
+        val navigationConfig = IssueNavigationConfiguration.getInstance(project)
+        val service = Env.data.services.getByKey("jira") ?: return
+        val url = service.url + "/browse/\$0"
+
+        val isExist = navigationConfig.links.any { it ->
```
Comment (Danil42Russia) [inline src/main/kotlin/com/vk/admstorm/AdmStartupService.kt:134]: `it` is not needed. The condition can be written in one line

File: src/main/kotlin/com/vk/admstorm/AdmStartupService.kt [hunk 4]
```diff
@@ -122,6 +126,21 @@ class AdmStartupService(private var project: Project) {
...
+            it.linkRegexp == url
+        }
+
+        if (!isExist) {
```
Comment (Danil42Russia) [inline src/main/kotlin/com/vk/admstorm/AdmStartupService.kt:138]: Don't you want to expand the condition?

```
if (navigationConfig.links.any { it.linkRegexp == url }) {
    return
}
```

File: src/main/kotlin/com/vk/admstorm/AdmStartupService.kt (hunk 3 above)
Comment (Hidanio) [reply src/main/kotlin/com/vk/admstorm/AdmStartupService.kt:134]: I prefer to always specify lambdas explicitly, but if it's important, then ok

File: src/main/kotlin/com/vk/admstorm/AdmStartupService.kt [hunk 5]
```diff
@@ -122,6 +126,18 @@ class AdmStartupService(private var project: Project) {
...
     }
 
+    private fun setupIssueTrackers(project: Project) {
+        val navigationConfig = IssueNavigationConfiguration.getInstance(project)
+        val service = Env.data.services.getByKey("jira") ?: return
+        val url = service.url + "/browse/\$0"
+
+        val isExist = navigationConfig.links.any { it.linkRegexp == url }
+        if (!isExist) {
+            val jiraLink = IssueNavigationLink("[A-Z]+\\-\\d+", url)
```
Comment (Danil42Russia) [inline src/main/kotlin/com/vk/admstorm/AdmStartupService.kt:136]: Let's use a regexp: `[A-Z]{2}[A-Z_0-9]*\-\d+`

Test it in code, please

**Review by Danil42Russia — APPROVED** [review]

**Pull request merged**